python/test/unit/language/test_core.py

```diff
@@ -2052,11 +2052,8 @@ def test_cast(dtype_x, dtype_z, bitcast, size, num_ctas, device):
         x_tri = torch.randn(size, dtype=torch.half, device=device).to(dtype=getattr(torch, dtype_x))
     else:
         x = numpy_random(size, dtype_str=dtype_x, low=-10, high=10) * 10
-        # Triton clamps negative values to zero, while numpy wraps around
-        # intmax, so avoid negatives for now.
-        # TODO: figure out which one should actually be happening, and test it
         if dtype_z in uint_dtypes:
-            x = np.absolute(x)
+            x = np.where(x >= 0, x, 0)
```

Comment:
Hi @peterbell10! This partially relates to https://github.com/triton-lang/triton/issues/7502. 

Previously, the test of converting negative numbers did not happen, I would like to see how it works for other backends. What you see in the code is the behavior on the xpu backend and it is definitely different from how it works in numpy for example. If all backends have the same behavior, I guess we can take it as expected and start to test it.

Reply:
all checks are successful. @ThomasRaoux @peterbell10 can we merge it?

Reply:
I didn't follow, what is the difference?